### PR TITLE
MODEXPW-493 - Outdated preview of matched records in case of remove (add) affiliation and upload the same file with Holdings, Items identifiers in Central tenant

### PR DESCRIPTION
**Pull request by alekGbuz**
[MODEXPW-493](https://folio-org.atlassian.net/browse/MODEXPW-493) - Outdated preview of matched records in case of remove (add) affiliation and upload the same file with Holdings, Items identifiers in Central tenant
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] Check logging
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -137,6 +137,10 @@
     {
       "id": "permissions",
       "version": "5.7"
+    },
+    {
+      "id": "consortia",
```
Comment (NikitaSedyx) [inline descriptors/ModuleDescriptor-template.json:142]: is consortia available for non-ecs?

Comment (alekGbuz) [reply descriptors/ModuleDescriptor-template.json:142]: hi @NikitaSedyx , the consortia logic is used as in mod-bulk-operations module. But the last one does not have consortia interface in module descriptor. May be it is better to use optional or remove it. Let's double check.

Comment (NikitaSedyx) [reply descriptors/ModuleDescriptor-template.json:142]: it must be optional, consortia is not part of platform - you had similar issue before, time to think what you include in required interfaces

Comment (NikitaSedyx) [reply descriptors/ModuleDescriptor-template.json:142]: additionally please create a PR to add interfaces in bulk ops

File: src/main/java/org/folio/dew/batch/bulkedit/jobs/BulkEditHoldingsProcessor.java [hunk 2]
```diff
@@ -126,38 +122,36 @@ private ExtendedHoldingsRecordCollection getHoldingsRecords(ItemIdentifier itemI
...
           try (var context = new FolioExecutionContextSetter(refreshAndGetFolioExecutionContext(tenantId, folioExecutionContext))) {
-            permissionsValidator.checkBulkEditReadPermissions(tenantId, EntityType.HOLDINGS_RECORD);
             var holdingsRecordCollection = getHoldingsRecordCollection(type, itemIdentifier);
             extendedHoldingsRecordCollection.getExtendedHoldingsRecords().addAll(
               holdingsRecordCollection.getHoldingsRecords().stream()
                 .map(holdingsRecord -> new ExtendedHoldingsRecord().tenantId(tenantId).entity(holdingsRecord)).toList()
             );
             extendedHoldingsRecordCollection.setTotalRecords(extendedHoldingsRecordCollection.getTotalRecords() + holdingsRecordCollection.getTotalRecords());
           } catch (Exception e) {
-            if (e instanceof FeignException && ((FeignException) e).status() == 401) {
```
Comment (khandramai) [inline src/main/java/org/folio/dew/batch/bulkedit/jobs/BulkEditHoldingsProcessor.java:139]: Do we really no longer need this logic, and should we not handle errors related to affiliations and permissions separately? This story is related to adding the Eureka integration but not to changing the integration with Okapi.

Comment (alekGbuz) [reply src/main/java/org/folio/dew/batch/bulkedit/jobs/BulkEditHoldingsProcessor.java:139]: Hi @khandramai , this logic was moved to tenantResolver.getAffiliatedPermittedTenantIds in the way to check all needed tenants - related to permissions and affiliations requirements.

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (alekGbuz) [reply descriptors/ModuleDescriptor-template.json]: Updated

File: src/main/java/org/folio/dew/batch/bulkedit/jobs/processidentifiers/ItemFetcher.java [hunk 3]
```diff
@@ -107,15 +110,9 @@ public synchronized ExtendedItemCollection process(ItemIdentifier itemIdentifier
               );
               extendedItemCollection.setTotalRecords(extendedItemCollection.getTotalRecords() + itemCollection.getTotalRecords());
             } catch (Exception e) {
-              if (e instanceof FeignException && ((FeignException) e).status() == 401) {
```
Comment (khandramai) [inline src/main/java/org/folio/dew/batch/bulkedit/jobs/processidentifiers/ItemFetcher.java:110]: Do we really need to remove logic related to handling affiliation errors? Especially, in logic related to Okapi.

**Comment by khandramai** [comment]
I assume we need to support logic for both Eureka and Okapi. Could you please double check if nothing necessary for the integration with Okapi has been removed.

File: src/main/java/org/folio/dew/batch/bulkedit/jobs/processidentifiers/ItemFetcher.java (hunk 3 above)
Comment (alekGbuz) [reply src/main/java/org/folio/dew/batch/bulkedit/jobs/processidentifiers/ItemFetcher.java:110]: Where were reported issues for affiliation checking, in the same time the such issues were not observed while checking affiliations using consortia functionality.

File: src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  private boolean isBulkEditReadPermissionExists(String tenantId, EntityType entityType) {
+    try {
+      return permissionsValidator.isBulkEditReadPermissionExists(tenantId, entityType);
+    } catch (FeignException e) {
+      throw new BulkEditException(e.getMessage());
+    }
+  }
+
+  protected String getAffiliationErrorPlaceholder(EntityType entityType) {
+    if (entityType == EntityType.ITEM) {
```
Comment (khandramai) [inline src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java:71]: To protect, please, throw unsupported exception for different to items and holdings cases.

File: src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      return NO_ITEM_AFFILIATION;
+    }
+    return NO_HOLDING_AFFILIATION;
+  }
+
+  protected String getViewPermissionErrorPlaceholder(EntityType entityType) {
+    if (entityType == EntityType.ITEM) {
```
Comment (khandramai) [inline src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java:78]: To protect, please, throw unsupported exception for different to items and holdings cases.

**Comment by alekGbuz** [comment]
> I assume we need to support logic for both Eureka and Okapi. Could you please double check if nothing necessary for the integration with Okapi has been removed.

Ok, lets double check it.

**Review by khandramai — APPROVED** [review]

File: src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java (hunk 5 above)
Comment (alekGbuz) [reply src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java]: Updated

File: src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java (hunk 4 above)
Comment (alekGbuz) [reply src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java]: Updated

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**